File: 조서영/1주차/1920.py
# 1920, Silver 4
# https://www.acmicpc.net/problem/1920

import sys

N=int(sys.stdin.readline())
A=sorted(list(map(int, sys.stdin.readline().rstrip().split())))
M=int(sys.stdin.readline())
llist=list(map(int, sys.stdin.readline().rstrip().split()))

def binary_search(S, target):
    high=len(S)-1
    low=0

    while low<=high:
        mid=(high+low)//2
        n=S[mid]

        if n==target:
            return 1
        elif n>target:
            high=mid-1
        elif n<target:
            low=mid+1
    
    return 0

for i in llist:
    print(binary_search(A, i))

File: 장익환/1주차/1920.cpp
/*
백준 1920번, https://www.acmicpc.net/problem/1920
*/

#include <iostream>
#include <algorithm>
using namespace std;

int main(void) {
	int N, M;
	int A[100000]; // N개의 정수 
	int key[100000]; // M개의 정수

	cin >> N;
	for (int i = 0; i < N; i++) {
		cin >> A[i];
	}

	cin >> M;
	for (int i = 0; i < M; i++) {
		cin >> key[i];
	}

	sort(A, A + N);

	for (int i = 0; i < M; i++) {
		// 이분 탐색으로 0인지 1인지 판정
		int left = -1;
		int right = N;
		int mid;

		while (left + 1 < right) {
			mid = (left + right) / 2;
			if (A[mid] < key[i]) {
				left = mid;
			}
			else {
				right = mid;
			}
		}
		if (A[right] == key[i]) {
			cout << "1\n";
		}
		else {
			cout << "0\n";
		}
	}
}
